cheroot/workers/monitor.py

```diff
@@ -0,0 +1,131 @@
+"""Monitor for threadpool."""
+__metaclass__ = type
+
+import threading
+import time
+
+from dynpool import DynamicPoolResizer
+
+
+class BackgroundTask(threading.Thread):
+    """A subclass of threading.Thread whose run() method repeats.
+
+    Use this class for most repeating tasks. It uses time.sleep() to wait
+    for each interval, which isn't very responsive; that is, even if you call
+    self.cancel(), you'll have to wait until the sleep() call finishes before
+    the thread stops. To compensate, it defaults to being daemonic, which means
+    it won't delay stopping the whole process.
+
+    :param int interval: time interval between invocation of input function
+    :param function: callable that needs to be invoked
+    :param args: list of arguments to pass to the function
+    :param kwargs: dictionary of keyword args to pass to the function
+    """
+
+    def __init__(self, interval, function, args=None, kwargs=None):
+        """Initialize a BackgroundTask."""
+        super(BackgroundTask, self).__init__()
+        self.interval = interval
+        self.function = function
+        self.args = args if args is not None else []
+        self.kwargs = kwargs if kwargs is not None else {}
+        self.running = False
+        self.daemon = True
+
+    def cancel(self):
+        """Cancel the task."""
+        self.running = False
+
+    def run(self):
+        """Invoke the callable with the given interval."""
+        self.running = True
+        while self.running:
+            time.sleep(self.interval)
```

Comment:
This inserts a delay before the first invocation. I think it'd be more intuitive to wait after it's called. Also, this would remove the need to do the following check.

Reply:
Not sure if I understand fully. Are you suggesting we do the following:
`
    def run(self):
        """Invoke the callable with the given interval."""
        self.running = True
        while self.running:
            # Any error in the calling function will be raised immediately
            self.function(*self.args, **self.kwargs)
            time.sleep(self.interval)
`

Reply:
Yes, that seems to be more responsive.

Reply:
Done